src/rgw/rgw_directory.h

```diff
@@ -0,0 +1,59 @@
+#ifndef CEPH_RGWDIRECTORY_H
+#define CEPH_RGWDIRECTORY_H
+
+#include <stdlib.h>
+#include <sys/types.h>
+#include <sstream>
+#include "rgw_common.h"
+#include "cpp_redis/cpp_redis"
+#include <string>
+#include <iostream>
+#include <vector>
+#include <list>
+#include <cstdint>
+#define dout_subsys ceph_subsys_rgw
+
+struct cache_obj {
+  std::string bucket_name; // s3 bucket name
+  std::string obj_name; //s3 obj name
+};
+
+struct cache_block {
+  cache_obj c_obj;
+  uint64_t size_in_bytes; // block size_in_bytes
+  std::vector<std::string> hosts_list; // list of hostnames <ip:post> of block locations
+};
+
+class RGWDirectory {
+  public:
+    RGWDirectory() {}
+    CephContext *cct;
+};
+
+class RGWBlockDirectory: RGWDirectory {
+  public:
+    RGWBlockDirectory() {}
+    RGWBlockDirectory(std::string blockHost, int blockPort):host(blockHost), port(blockPort) {}
+    
+    void init(CephContext *_cct) {
+      cct = _cct;
+      host = cct->_conf->rgw_directory_host;
+      port = cct->_conf->rgw_directory_port;
+    }
+	
+    void findClient(cpp_redis::client *client);
```

Comment:
`client` is already a member variable, so it's redundant to pass it to this member function. As a result, should I remove this parameter? The only use I can see it having is if `findClient` will be called to find connected clients that do not belong to any `RGWBlockDirectory` instance.